### PR TITLE
Improve linting when third party libs have been linted

### PR DESCRIPTION
**Pull request by borkdude**
Mainly improvements regarding clojurewerkz/quartzite. This fixes all warnings after all deps have been analyzed (e.g. using clojure-lsp).

/cc @dpsutton 

This is a follow up PR to https://github.com/metabase/metabase/pull/20208, so please squash/merge that one first and then squash/merge this one.

### CODE REVIEW

**Review by dpsutton — APPROVED** [review]
@borkdude what's the story on excluding the clojurescript from specmontash. I'm surprised it came up in this patch but 100% love that we are excluding that and wish we had done that earlier.

**Comment by borkdude** [comment]
@dpsutton It pulled in a very old version of ClojureScript which gave a false positive for calling `clj->js` with three arguments: `(clj->js x :keywordize-keys true)`

File: src/metabase/task/follow_up_emails.clj [hunk 1]
```diff
@@ -66,7 +66,7 @@
 (def ^:private follow-up-emails-trigger-key "metabase.task.follow-up-emails.trigger")
 
 (defmethod task/init! ::SendFollowUpEmails [_]
-  (let [job     (jobs/build
+  (let [job    (jobs/build
```
Comment (noahmoss) [inline src/metabase/task/follow_up_emails.clj:69]: tiny nit: is there a reason for this whitespace change? it's no longer in line with the expression below

Comment (borkdude) [reply src/metabase/task/follow_up_emails.clj:69]: I'll fix that. I probably did this to trigger the linting again :).

**Comment by borkdude** [comment]
Correction to the above, CLJS was pulled it by a dependency of loom:

```
reifyhealth/specmonstah 2.0.0
  . better-cond/better-cond 2.0.1-20170618.063731-4
  . aysylu/loom 1.0.2
    X org.clojure/data.priority-map 0.0.5 :superseded
    . tailrecursion/cljs-priority-map 1.2.1
      . org.clojure/clojurescript 1.7.170
```

File: src/metabase/task/follow_up_emails.clj (hunk 1 above)
Comment (borkdude) [reply src/metabase/task/follow_up_emails.clj]: Fixed.

**Pull request merged**